### ISSUE TITLE
Original Entity Scores

### ISSUE THREAD

**Issue by peluz** [eraldoluis/LeNER-Br#1]
Hello, Eraldo Fernandes!

This is Pedro Luz, the first author of the LeNER-Br paper. First, I would like to thank you for the interest in our project. It's very nice to see our work being explored by others.

We only report per-token scores in the original paper due to the fact that we were comparing with a work that used the Paramopama Dataset, which does not use a tagging scheme such as IOB or IOBES; thus, no mention boundary annotation is available for per-entity evaluation .

That said, we did compute entity scores for LeNER-Br and [reported them](https://cic.unb.br/~teodecampos/LeNER-Br/luz_etal_propor2018_slides.pdf) at the Propor 2018 conference.

Best regards,

Pedro Luz

**Comment by eraldoluis** [comment]
Oi Pedro. Obrigado pelo contato. Eu não tinha visto os slides. Foi bom você apontar estes resultados pois um aluno meu está concluindo a escrita da sua dissertação e vamos incluí-los (estávamos usando um resultado que nós geramos usando seu código).

Me permita fazer um comentário sobre o esquema de anotação do Paramopama. Realmente, ele não usa IOB. Na literatura de NER, existem alguns corpora que seguem este esquema. Isto é possível em muitos corpora de NER pois é raro (em alguns casos nunca ocorre) duas entidades estarem "coladas" uma na outra (e ainda terem o mesmo tipo). Este é o único caso que gera problemas para este esquema de anotação. Existem outros aspectos a serem considerados como, por exemplo, qual o impacto do esquema no desempenho de classificadores. De qualquer maneira, é sempre possível converter de um esquema para outro.

Só queria chamar a atenção que o esquema de anotação não determina a métrica usada. Inclusive, apesar de não estar explícito no artigo do Paramopama, eles usaram a métrica por entidade. Na realidade, pelo menos desde a CoNLL-2002 (quando comecei a acompanhar a literatura de NER), eu não conheço nenhum trabalho que usa a métrica por token. Muitos artigos nem mencionam que estão usando a métrica por entidade, pois este é o padrão vastamente usado na literatura. Quando lemos o trabalho de vocês, assumimos que seguiam este padrão. Quando obtivemos os primeiros resultados é que começamos a desconfiar e, posteriormente, confirmamos pelo seu código.

**Comment by peluz** [comment]
Obrigado pela resposta. 

Fico feliz que o trabalho tenha sido útil para a dissertação do seu aluno.

Eu gostaria de esclarecer que o artigo compara os resultados no Paramopama com o trabalho [Uma arquitetura híbrida LSTM-CNN para reconhecimento de entidades nomeadas em textos naturais em língua portuguesa](https://docplayer.com.br/122820147-Uma-arquitetura-hibrida-lstm-cnn-para-reconhecimento-de-entidades-nomeadas-em-textos-naturais-em-lingua-portuguesa.html). A seção de métrica do referido trabalho indica que o F1 Score foi calculado por palavras e não entidades. Isso se reflete nos resultados, que foram bons demais para ser por entidade, considerando que é uma arquitetura simples que sequer usa CRF.

Dito isso, é certo que o mais adequado é calcular resultados considerando toda a entidade. Mas mantenho minha dúvida de como fazê-lo em casos que não adotam tagging scheme. Simplesmente assumir que entidades do mesmo tipo e adjacentes se referem a mesma entidade? Dada a raridade que você mencionou penso que funcionaria, mas me deixa "um gosto ruim na boca" hahaha.

**Comment by eraldoluis** [comment]
> Eu gostaria de esclarecer que o artigo compara os resultados no Paramopama com o trabalho

Sim. Sua comparação foi justa. Só não seguiu a métrica mais usada na literatura.

> Mas mantenho minha dúvida de como fazê-lo em casos que não adotam tagging scheme. 
> Simplesmente assumir que entidades do mesmo tipo e adjacentes se referem a mesma entidade? 

Isso mesmo. Mas isto não seria assumir algo novo. Seria assumir que o foi assumido por quem anotou o corpus. É esta a suposição deste esquema de anotação.

> Dada a raridade que você mencionou penso que funcionaria, 
> mas me deixa "um gosto ruim na boca" hahaha.

:) Eu entendo. Mas para NER, esta suposição realmente não tem impacto relevante na maioria dos corpora. Pra você tirar este gosto da boca :), busque quantas ocorrências existem no LeNER-Br (ou no HAREM) de duas entidades consecutivas do mesmo tipo. Aposto que dá pra contar nos dedos de uma mão, se é que você vai encontrar alguma.

**Comment by peluz** [comment]
> Isso mesmo. Mas isto não seria assumir algo novo. Seria assumir que o foi assumido por quem anotou o corpus. É esta a suposição deste esquema de anotação.

Entendi, assim fica mais pacificado pra mim.
